### PR TITLE
Leverage on dpctl.tensor implementation in `dpnp.count_nonzero`

### PR DESCRIPTION
**Pull request by antonwolfy**
The PR proposes to leverage on dpctl.tensor implementation in `dpnp.count_nonzero` function.
Additional missing description of input parameters is added.

Also existing helper `dpnp.get_result_array` function is extended to allow `usm_ndarray` as input array.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_iface_statistics.py [hunk 1]
```diff
@@ -904,11 +904,9 @@ def std(
...
         dpnp.sqrt(result, out=result)
     else:
-        dpt_array = dpnp.get_usm_ndarray(a)
-        result = dpnp_array._create_from_usm_ndarray(
-            dpt.std(dpt_array, axis=axis, correction=ddof, keepdims=keepdims)
-        )
-        result = dpnp.get_result_array(result, out)
+        usm_a = dpnp.get_usm_ndarray(a)
+        usm_res = dpt.std(usm_a, axis=axis, correction=ddof, keepdims=keepdims)
+        result = dpnp.get_result_array(usm_res, out)
```
Comment (vlad-perevezentsev) [inline dpnp/dpnp_iface_statistics.py:909]: Сould you change `def mean()` as well?
For example
```
def mean(a, /, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
...
    dpt_array = dpnp.get_usm_ndarray(a)
    result =  dpt.mean(dpt_array, axis=axis, keepdims=keepdims)
    result = dpt.astype(result, dtype) if dtype is not None else result

    return dpnp.get_result_array(result, out, casting="same_kind")
```

File: dpnp/dpnp_iface_counting.py [hunk 2]
```diff
@@ -44,25 +44,38 @@
 __all__ = ["count_nonzero"]
 
 
-def count_nonzero(a, axis=None, *, keepdims=False):
+def count_nonzero(a, axis=None, *, keepdims=False, out=None):
```
Comment (vlad-perevezentsev) [inline dpnp/dpnp_iface_counting.py:47]: Could you add tests covering `out` parameter?

Comment (vlad-perevezentsev) [reply dpnp/dpnp_iface_counting.py:47]: and `keepdims` as well

Comment (antonwolfy) [reply dpnp/dpnp_iface_counting.py:47]: Done

**Review by vlad-perevezentsev — APPROVED** [review]
Looks good!
Thank you @antonwolfy

**Pull request merged**